tests/test_service.py

```diff
@@ -830,3 +830,183 @@ def testGetAlertWithMultipleTriggers(self):
+
+
+    @mock.patch('requests.Session.post', return_value=MockResponse(json.dumps(compalert_D), 200))
+    def testAddChildAlert(self, mockPost):
+        alert = Alert.from_dict(compalert_D)
+        self.assertTrue(isinstance(alert, Alert))
+        delattr(alert, "id")
+        comp_alert = self.argus.alerts.create_composite_alert(alert)
+        self.assertTrue(isinstance(comp_alert, Alert))
+
+        with mock.patch('requests.Session.post', return_value=MockResponse(json.dumps(childAlert_1), 200)):
+            child_alert = self.argus.alerts.add_child_alert_to_composite_alert(comp_alert.id, Alert.from_dict(childAlert_1))
+        self.assertEqual(child_alert.alertType, 'COMPOSITE_CHILD')
+        self.assertTrue(isinstance(child_alert, Alert))
+
+    @mock.patch('requests.Session.post', return_value=MockResponse(json.dumps(compalert_D), 200))
+    def testAddTriggerToChildAlert(self, mockPost):
+        alert = Alert.from_dict(compalert_D)
+        self.assertTrue(isinstance(alert, Alert))
+        delattr(alert, "id")
+        comp_alert = self.argus.alerts.create_composite_alert(alert)
+        self.assertTrue(isinstance(comp_alert, Alert))
+
+        with mock.patch('requests.Session.post', return_value=MockResponse(json.dumps(childAlert_1), 200)) as mock_add_childalert:
+            child_alert = self.argus.alerts.add_child_alert_to_composite_alert(comp_alert.id,
+                                                                               Alert.from_dict(childAlert_1))
+            self.assertEqual(child_alert.alertType, 'COMPOSITE_CHILD')
+            self.assertTrue(isinstance(child_alert, Alert))
+            call_args = tuple(mock_add_childalert.call_args)
+            uri_path = os.path.join(endpoint, "alerts/{}/children".format(comp_alert.id))
+            self.assertIn((uri_path,), call_args)
+
+        with mock.patch('requests.Session.post', return_value=MockResponse(json.dumps(childAlert_trigger_1), 200)) as mock_trigger_post:
+            trigger = self.argus.alerts.add_trigger_to_child_alert(child_alert.id, Trigger.from_dict(childAlert_trigger_1))
+            self.assertTrue(isinstance(trigger, Trigger))
+            call_args = tuple(mock_trigger_post.call_args)
+            uri_path = os.path.join(endpoint, "alerts/{}/triggers".format(child_alert.id))
+            self.assertIn((uri_path,), call_args)
+
+    def testAddNotification(self):
+        with mock.patch('requests.Session.post', return_value=MockResponse(json.dumps(compalert_D), 200)) as mock_add_comp_alert:
+            alert = Alert.from_dict(compalert_D)
+            self.assertTrue(isinstance(alert, Alert))
+            delattr(alert, "id")
+            comp_alert = self.argus.alerts.create_composite_alert(alert)
+            self.assertTrue(isinstance(comp_alert, Alert))
+            call_args = mock_add_comp_alert.call_args
+            uri_path = os.path.join(endpoint, "alerts")
+            self.assertIn((uri_path,), call_args)
```

Comment:
These also appear in several methods. Can you move this to a helper method?

Reply:
Fixed code.  Move common functionality into a helper

Reply:
Hm.. I don't see these moved into a helper method.. I'm talking about lines like these:
            call_args = tuple(mock_trigger_post.call_args)
            uri_path = os.path.join(endpoint, "alerts/{}/triggers".format(child_alert.id))
            self.assertIn((uri_path,), call_args)

Reply:
@j-ma-sf they vary from function to function and we need to create more helper functions for each one of them.  We are not going to gain much out of creating so many functions

Reply:
Oh I see, yeah